### ISSUE TITLE
Apache Pinot facing issues with Helix version 1.4.0

### ISSUE THREAD

**Issue by abhioncbr** [apache/helix#2850]
### Describe the bug
Apache Pinot is using Helix for various cluster management tasks. I recently tried to upgrade the Helix dependency from 1.3.1 to 1.4.0; however, we are seeing a lot of test failures and startup failures with the newer version of Helix. 

Based on the investigation, I pin down the issues because of the change in this [PR](https://github.com/apache/helix/pull/2782/files#diff-321cf13f9b2ecee61b329273b373b7cac7ec1b2e4b99b66c3d045847f2c1f040L94).

I am trying to understand the change introduced in the PR mentioned above and workaround for the issue. For reference(Apache Pinot Helix upgrade [PR](https://github.com/apache/pinot/pull/13486))

cc: @jackjlli / @Jackie-Jiang

**Comment by junkaixue** [comment]
@abhioncbr did Pinot use WAGED rebalance? This WAGED related changes.

Original behavior is when new resource added without partition can cause NPE and break the Helix pipeline. This PR fix this behavior.

adding the author : @himanshukandwal

**Comment by soumitra-st** [comment]
How can we get a closure on this issue? I want to upgrade Pinot to [Change tagged job assignment behavior](https://github.com/apache/helix/pull/2976) fix to improve the Minions.

**Comment by kishoreg** [comment]
> @abhioncbr did Pinot use WAGED rebalance? This WAGED related changes.
> 
> Original behavior is when new resource added without partition can cause NPE and break the Helix pipeline. This PR fix this behavior.
> 
> adding the author : @himanshukandwal

This PR seems like a regression. The change seems to be in resourcecomputationstage that is applicable to all rebalancers and is not limited to waged rebalancers. Can we please revert this?

**Comment by npawar** [comment]
@junkaixue checking if you got a chance to look at this? We want to upgrade in PInot but this will block us..

**Comment by junkaixue** [comment]
Yeah. The place to fix may not proper. I will revert this change and ask the author do the proper fix for that.

For the proper NPE, it should not just filter the thing out without finding the right place.

**Comment by junkaixue** [comment]
@himanshukandwal @xyuanlu This change is reverted: https://github.com/apache/helix/pull/2995

Reopen the ticket: https://github.com/apache/helix/issues/2781

@npawar @kishoreg 

Close this ticket.
